src/main/java/com/conveyal/gtfs/loader/JdbcGtfsLoader.java

```diff
@@ -385,7 +385,17 @@ private int loadInternal (Table table) throws Exception {
                 String string = csvReader.get(f);
                 // Use spec table to check that references are valid and IDs are unique.
                 Set<NewGTFSError> errors = table.checkReferencesAndUniqueness(keyValue, lineNumber, field, string, referenceTracker);
-                errorStorage.storeErrors(errors);
+                if (
+                    table.name.equals("calendar_dates") &&
+                    "1".equals(csvReader.get(Field.getFieldIndex(fields, "exception_type")))
+                ){
+                    // Do not record bad reference errors for calendar date entries that add service
+                    // (exception type=1) because a corresponding service_id in calendars.txt is not required in
+                    // this case.
+                    LOG.info("Skipping bad reference for calendar_date adding service (service_id={}).", keyValue);
```

Comment:
Are we absolutely sure that no other errors are skipped here? Maybe to future-proof this, it'd be a good idea to explicitly check if the exact error we're looking for occurred and only then do this thing. Also, I'd reword the log statement to not say that it's skipping something bad, but maybe something like an added service calendar_date entry was found that doesn't need to have an entry in the calendar file.

Reply:
I agree on both points - good to check for a specific error type and make messages as clear as we can.